lib/core/src/lib/datatable/components/datatable/datatable.component.spec.ts

```diff
@@ -415,6 +415,24 @@ describe('DataTable', () => {
         expect(dataTable.resetSelection).toHaveBeenCalled();
     });
 
+    it('should reset selection on multiselect change', () => {
+        spyOn(dataTable, 'resetSelection').and.callThrough();
+
+        dataTable.data = new ObjectDataTableAdapter([{ name: '1' }, { name: '2' }], [new ObjectDataColumn({ key: 'name' })]);
+        const rows = dataTable.data.getRows();
+        rows[0].isSelected = true;
+        rows[1].isSelected = true;
+
+        expect(rows[0].isSelected).toBeTruthy();
+        expect(rows[1].isSelected).toBeTruthy();
+
+        dataTable.ngOnChanges({
+            multiselect: new SimpleChange(true, false, false)
+        });
+
+        expect(dataTable.resetSelection).toHaveBeenCalled();
```

Comment:
This kind of unit test is testing the flow rather than the behaviour. 
Instead of checking a method is called you should check the selection is empty.